document-generator-api/src/main/java/uk/gov/companieshouse/document/generator/api/service/DocumentTypeService.java

```diff
@@ -0,0 +1,14 @@
+package uk.gov.companieshouse.document.generator.api.service;
+
+import uk.gov.companieshouse.document.generator.api.Exception.DocumentGeneratorServiceException;
+
+public interface DocumentTypeService {
+
+    /**
+     * get Document type by pattern matching the resourceUri to a regex pattern
```

Comment:
Should we be exposing the implementation detail in the Javadoc? May be considered knowledge bleed.